---
title: "**Transcriptional and Translational Dynamics of Zika and Dengue Virus Infection**"
author: "Md Shazid Hasan, Dylan Harris, Jahanara Sonia, Weixia Deng"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE)
```

# Background and Introduction:

Translation, the key event in all living things in which the genetic code is decoded as proteins are synthesized, has been underappreciated as a regulatory stage of gene expression. Historically, gene expression monitoring has focused on the levels of mRNA (the transcriptome) determined by methods that continue to be more informative, such as the widely used high throughput RNA-sequencing (RNA-seq) that alone is not sufficient to determine protein expression level. Proteomics study can address this questions regarding protein expression, but measuring protein levels combined with RNA-seq provides only an indirect measure of translation efficiency and does not account for differences in protein stability. To study and understand translation regulation, ribosome profiling is one of the powerful tools. Ribosome Profiling or RiboSeq is a RNAseq-based powerful deep sequencing technique that sequences the RNA occupied by translating ribosome instead of sequencing the total mRNA.  This can provide a panaroma of whole translation process.

In this study, the authors have used ribosome profiling to understand translation regulation in virus infection. Zika virus (ZIKV) and dengue virus (DENV) belongs to the family Flaviviridae and contain single-stranded plus-sense RNA genomes. Both viruses are transmitted to humans by tropical Aedes mosquitos that are increasingly reaching first world regions. ZIKV and DENV share over 90% of their genome sequences. Also, previous studies on the transcriptional and immunological effects of DENV and ZIKV have revealed that both viruses induce a classical Type I interferon anti-viral response. In this study, they have done high-resolution ribosome profiling and RNA deep sequencing (RNA-seq) to define the gene expression and mRNA translation dynamics of the viral and host genomes during ZIKV and DENV infection of human neuronal progenitor cells (hNPCs). Their data highlights the cellular stress response and the activation of RNA translation and polyamine metabolism during DENV and ZIKV infection.

In this project, we tried to reproduce and understand the differentially expressed gene analysis from this study, both in transcriptional and translational level.

# Overall design
They performed ribosome footprinting and RNA sequencing on human neuronal progenitor cells (hNPCs) cells either uninfected or infected with ZIKV or DENV in two biological replicates that were collected 72 hours post infection. 

# Source of the raw data

[**Data Link**](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE207347)

In the source, unifected samples are indicated as A1 and B1, ZIKV infected samples are indicated as A2, B2 and DENV infected samples are indicated as A3 and B3. Then the raw sequences were processed.

# Technical Details:

We were able to replicate most figures that did not require a reference genome using `ggplot2` 3.4.2 in `R` version 4.3.0.

# Difficulties
-	Rerunning analysis leads to differing adjusted p values.
	Different statistical software was used by the Singh et al. meaning a different algorithm is used to correct false discovery rate.
-	Some rounding was used by the paper resulting in slight differences in figures even when their p adjusted was used
-	A large portion of the figures in this paper are generated through Ensembl which compares the RNA sequences to the human genome giving them labels
-	No documentation makes this difficult

```{r}
# Load packages and read Files
library(tidyverse)
library(data.table)
library(scales)
library(metR)

df.ZIKV <- fread("data/GSE207347_A1B1_vs_A2B2_ZIKV_ribodiff_name.txt.gz")
df.DENV <- fread("data/GSE207347_A1B1_vs_A3B3_DENV_ribodiff_name.txt.gz")
df.ZIKV_DE <- fread("data/GSE207347_ZIKV_DESeq2_result_name.txt.gz")
df.DENV_DE <- fread("data/GSE207347_DENV_DESeq2_result_name.txt.gz")
```

# Figure 1 B and C

```{r, fig.align = 'center', fig.show = 'hold', out.width = '49%'}
# Figure 1 B
fig1b <- ggplot(subset(df.ZIKV_DE, !is.na(padj)),
                aes(log2FoldChange, -log10(padj)))

fig1b +
  geom_point(aes(colour = -log10(padj) > -log10(0.05)), show.legend = FALSE) +
  labs(title = "ZIKV") + ylab("p adj(-log10)") +
  xlab("RNA log2Fold change \n (Infected vs uninfected)")  +
  geom_vline(xintercept = 0) + geom_hline(yintercept = -log10(0.05)) +
  scale_color_manual(values = c('black', 'red')) +
  annotate("text", x = -1.5, y = 20,
           label = "Down regulated \n q<0.05 \n (n=335)",
           col = "black") +
  annotate("text", x = 1.5, y = 20,
           label = "Up regulated \n q<0.05 \n (n=445)",
           col = "black") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.background = element_blank(),
        plot.title = element_text(hjust = 0.5),
        axis.line = element_line(colour = "black"),
        panel.border = element_rect(colour = "black",
                                    fill = NA, linewidth = 1.5))

# Figure 1 C
fig1c <- ggplot(subset(df.DENV_DE, !is.na(padj)),
                aes(log2FoldChange, -log10(padj)))

fig1c +
  geom_point(aes(colour = -log10(padj) > -log10(0.05)), show.legend = FALSE) +
  labs(title = "DENV") + ylab("p adj(-log10)") +
  xlab("RNA log2Fold change \n (Infected vs uninfected)") +
  geom_vline(xintercept = 0) + geom_hline(yintercept = -log10(0.05)) +
  scale_color_manual(values = c('black', 'red')) +
  annotate("text", x = -1.5, y = 14,
           label = "Down regulated \n q<0.05 \n (n=37)",
           col = "black") +
  annotate("text", x = 2, y = 14,
           label = "Up regulated \n q<0.05 \n (n=156)",
           col = "black") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.background = element_blank(),
        plot.title = element_text(hjust = 0.5),
        axis.line = element_line(colour = "black"),
        panel.border = element_rect(colour = "black",
                                    fill = NA, linewidth = 1.5))
```

```{r}
df.ZIKV_DE.sub <- subset(df.ZIKV_DE, !is.na(padj))
# Count of Up regulated with q < 0.05
# (df.ZIKV_DE.sub$log2FoldChange > 0 & df.ZIKV_DE.sub$padj < 0.05) %>% sum()
# Count of Down regulated with q < 0.05
# (df.ZIKV_DE.sub$log2FoldChange < 0 & df.ZIKV_DE.sub$padj < 0.05) %>% sum()
```


```{r}
df.DENV_DE.sub <- subset(df.DENV_DE, !is.na(padj))
# Count of Up regulated with q < 0.05
# (df.DENV_DE.sub$log2FoldChange > 0 & df.DENV_DE.sub$padj < 0.05) %>% sum()
# Count of Down regulated with q < 0.05
# (df.DENV_DE.sub$log2FoldChange < 0 & df.DENV_DE.sub$padj < 0.05) %>% sum()
```


Figure 1B & C: To reproduce Figure 1B and C we used log2fold change in RNA expression (uninfected vs infected) in the X axis and p(adjusted value) in Y axis. We found upregulation of 445 mRNAs and downregulation of 335 mRNAs in case of ZIKV infection. Also, we found upregulation of 156 mRNAs and downregulation of 37mRNAs in DENV infection, which is the same as the study.

The x-axis is `log2FoldChange` and y-axis is `padj` from `GSE207347_ZIKV_DESeq2_result_name.txt.gz` and `GSE207347_DENV_DESeq2_result_name.txt.gz` files.

# Figure 2 A and B

```{r, fig.align = 'center', fig.show = 'hold', out.width = '49%'}
# Figure 2 A

ZIKV.up <- df.ZIKV_DE.sub$log2FoldChange > 0 & df.ZIKV_DE.sub$padj < 0.05
df.ZIKV_DE.up <- df.ZIKV_DE.sub[ZIKV.up,]

fig2a <- ggplot(subset(df.ZIKV_DE.up, !is.na(padj)),
                aes(x = -log10(padj), y = log2FoldChange))

fig2a +
  geom_point(aes(colour = -log10(padj) > -log10(0.001)), show.legend = FALSE) +
  xlim(0, 20) + ylim(0, 5) + labs(title = "ZIKV Upregulated RNA") +
  xlab("p adj(-log10)") + 
  ylab("Up regulated RNA \n log2Fold change \n (Infected vs uninfected)") +
  scale_color_manual(values = c('black', 'red')) +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.background = element_blank(),
        plot.title = element_text(hjust = 0.5),
        axis.line = element_line(colour = "black"))

# Count of q < 0.001
# sum(df.ZIKV_DE.up$padj < 0.001)

# Figure 2 B

DENV.up <- df.DENV_DE.sub$log2FoldChange > 0 & df.DENV_DE.sub$padj < 0.05
df.DENV_DE.up <- df.DENV_DE.sub[DENV.up,]

fig2b <- ggplot(subset(df.DENV_DE.up, !is.na(padj)),
                aes(x = -log10(padj), y = log2FoldChange))

fig2b +
  geom_point(aes(colour = -log10(padj) > -log10(0.001)), show.legend = FALSE) +
  xlim(0, 20) + ylim(0, 5) + labs(title = "DENV Upregulated RNA") +
  xlab("p adj(-log10)") + 
  ylab("Up regulated RNA \n log2Fold change \n (Infected vs uninfected)") +
  scale_color_manual(values = c('black', 'red')) +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.background = element_blank(),
        plot.title = element_text(hjust = 0.5),
        axis.line = element_line(colour = "black"))

# Count of q < 0.001
# sum(df.DENV_DE.up$padj < 0.001)
```

Figure 2A & 2B: Here, we tried to analyze the upregulated RNA in ZIKV(fig. 2A) and DENV (fig. 2B) infected samples compared to the uninfected samples, using DSeq2 processed data. Although, in their plots they highlighted the genes that are upregulated both in ZIKV and DENV infection, our plot resembles their plot showing 104 (ZIKV infection) and 44 (DENV infection) upregulated RNAs.

The x-axis is `padj` and y-axis is `log2FoldChange` > 0 from `GSE207347_ZIKV_DESeq2_result_name.txt.gz` and `GSE207347_DENV_DESeq2_result_name.txt.gz` files.

# Figure 3 A and B

```{r, fig.align = 'center', fig.show = 'hold', out.width = '49%'}
# Figure 3 A

colnames(df.ZIKV)[7] <- "log2FC_TE_DrugTreated_vs_Control"

df.ZIKV.sub <- subset(df.ZIKV, !is.na(padj))
df.ZIKV.sub <- df.ZIKV.sub[df.ZIKV.sub$padj < 0.5,]

fig3a <- ggplot(df.ZIKV.sub,
                aes(x = log2FC_TE_DrugTreated_vs_Control, y = padj))

fig3a +
  geom_point(aes(color = padj > 0.05), show.legend = FALSE) +
  scale_y_continuous(trans = reverselog_trans(base = 10),
                     breaks = c(1, 0.1, 0.01, 0.001)) +
  scale_x_continuous(limits = c(-8, 8),
                     breaks = seq(-8, 8, 2)) +
  scale_color_manual(values = c('red', 'black')) +
  xlab("log2FC_TE(ZIKV infected vs control)") + 
  ylab("q adj(-log10)") +
  geom_hline(yintercept=0.05, linetype=2, col="black") +
  geom_vline(xintercept=0, linetype=2, col="black") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.background = element_blank(),
        plot.title = element_text(hjust = 0.5),
        axis.line = element_line(colour = "black"),
        panel.border = element_rect(colour = "black",
                                    fill = NA, linewidth = 1))

# Figure 3 B

colnames(df.DENV)[7] <- "log2FC_TE_DrugTreated_vs_Control"

df.DENV.sub <- subset(df.DENV, !is.na(padj))
df.DENV.sub <- df.DENV.sub[df.DENV.sub$padj < 0.5,]

fig3b <- ggplot(df.DENV.sub,
                aes(x = log2FC_TE_DrugTreated_vs_Control, y = padj))

fig3b +
  geom_point(aes(color = padj > 0.05), show.legend = FALSE) +
  scale_y_continuous(trans = reverselog_trans(base = 10),
                     labels = trans_format("log10", math_format(10^.x)),
                     breaks = trans_breaks("log10", function(x) 10^x)) +
  scale_x_continuous(limits = c(-8, 8),
                     breaks = seq(-8, 8, 2)) +
  scale_color_manual(values = c('red', 'black')) +
  xlab("log2FC_TE(DENV infected vs control)") + 
  ylab("q adj(-log10)") +
  geom_hline(yintercept=0.05, linetype=2, col="black") +
  geom_vline(xintercept=0, linetype=2, col="black") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.background = element_blank(),
        plot.title = element_text(hjust = 0.5),
        axis.line = element_line(colour = "black"),
        panel.border = element_rect(colour = "black",
                                    fill = NA, linewidth = 1))
```

Figure 3A and 3B: These figures indicate translational changes induced by ZIKV and DENV infection. Here, to reproduce the data we have used the ribodiff dataset where we have used log2 fold changes in Translational Efficiency in X axis. We also had to cut off all the p values greater than 0.05. For Y axis, q adusted value is used where we had to use reverse log10 scale to model the graph the same as the study's.

The x-axis is `log2FC_TE_DrugTreated_vs_Control` and y-axis is `padj` < 0.5 from `GSE207347_A1B1_vs_A2B2_ZIKV_ribodiff_name.txt.gz` and `GSE207347_A1B1_vs_A2B2_DENV_ribodiff_name.txt.gz` files.

```{r}
rm(list = ls())
```

# Figure 2 E

```{r}
#Load data and packages
library(tidyverse)
library(data.table)
Ribodiff.ZIKV <- fread("data/GSE207347_A1B1_vs_A2B2_ZIKV_ribodiff_name.txt.gz")
Ribodff.DENV <- fread("data/GSE207347_A1B1_vs_A3B3_DENV_ribodiff_name.txt.gz")
DESeq2.ZIKV <- fread("data/GSE207347_ZIKV_DESeq2_result_name.txt.gz")
DESeq2.DENV <- fread("data/GSE207347_DENV_DESeq2_result_name.txt.gz")
```

```{r, fig.align = 'center', fig.show = 'hold', out.width = '60%'}
#Grabbing tRNA Synthetases
ZIKV.trna <- subset(DESeq2.ZIKV, na %in% c("CARS", "YARS", "WARS", "AARS", "MARS", "SARS", "GARS", "VARS", "TARS", "IARS"))
DENV.trna <- subset(DESeq2.DENV, na %in% c("CARS", "YARS", "WARS", "AARS", "MARS", "SARS", "GARS", "VARS", "TARS", "IARS"))

#Combining ZIKV and DENV data
ZIKV.trna$virus <- "ZIKV"
DENV.trna$virus <- "DENV"
complete.trna <- rbind(ZIKV.trna, DENV.trna)

ggplot(complete.trna, aes(x=na, y=log2FoldChange, label=log2FoldChange)) +
  geom_bar(stat = "identity", aes(fill=virus), width=0.75, show.legend = FALSE) +
  ggtitle("Aminoacyl tRNA Sythetases") +
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1,), 
        plot.title = element_text(hjust = 0.5),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.background = element_blank(),
        axis.line = element_line(colour = "black")) +
  xlab("Gene Name") + ylab("RNA log2 Fold Change") +
  facet_wrap(~virus)
#So weirdly, the paper shows TARS decreasing for DENV, but examining the data shows otherwise? Maybe they used a different variant of TARS i.e. TARSL2?
#Also can't seem to adjust color of the bars.
```

```{r}
rm(list = ls())
```

Figure 2E: Here, we analyzed RNA fold change of ZIKV and DENV induced aminoacyl tRNA synthetases (ARS) genes. Interestingly, here, our plot showed difference compared to the actual study. For example, TARS genes showed non-significant decrease in RNA abundance in their study, while ours still shows an increase which is also indicated by their data tables.

# Test (Figure 1 F)

```{r}
# Some test, TL;DR
library(tidyverse)
library(data.table)
library(clusterProfiler)
library(enrichplot)

# Read Files
df.ZIKV <- fread("data/GSE207347_A1B1_vs_A2B2_ZIKV_ribodiff_name.txt.gz")
df.ZIKV_DE <- fread("data/GSE207347_ZIKV_DESeq2_result_name.txt.gz")

df.ZIKV_DE.sub <- subset(df.ZIKV_DE, !is.na(padj))
idx <- df.ZIKV_DE.sub$log2FoldChange < 0 & df.ZIKV_DE.sub$padj < 0.05
df.sub <- df.ZIKV_DE.sub[idx,]

# ZIKV_DE Data

original_gene_list <- df.sub$log2FoldChange
names(original_gene_list) <- df.sub$ID
gene_list <- na.omit(original_gene_list)
gene_list <- sort(gene_list, decreasing = TRUE)

organism <- "org.Hs.eg.db"
library(organism, character.only = TRUE)
# keytypes(org.Hs.eg.db)

gse <- gseGO(geneList=gene_list, 
             ont ="ALL", 
             keyType = "ENSEMBL",
             pvalueCutoff = 0.05, 
             verbose = TRUE, 
             OrgDb = organism, 
             pAdjustMethod = "none") %>% pairwise_termsim()


emapplot(gse)
```


Figure 1F: This figure is the most interesting part of our analysis. Here we tried to reproduce their SRING analysis of gene clusters to understand interaction between downregulated genes. Although we ended up producing a very different figure, the clusters we got were similar to the study's. Their study showed clusters of genes associated with cell cycle and histone regulation, where the clusters we found also showed clusters associated with cell cycle and chromatin organization.
